### PR TITLE
CI: Sync Minimum Versions

### PR DESCRIPTION
**Pull request by lithomas1**
https://pandas.pydata.org/docs/getting_started/install.html#dependencies

cc @simonjayhawkins

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -22,13 +22,15 @@ jobs:
         py_3.7_32:
           PYTHON_VERSION: "3.7"
           PYTHON_ARCH: "x86"
-          NP_BUILD_DEP: "1.16.5"
-          NIGHTLY_BUILD: "true"
+          NP_BUILD_DEP: "1.17.3"
```
Comment (simonjayhawkins) [inline azure-pipelines.yml:25]: we still need to build 1.3.1, 1.3.2, .... etc 

we won't be increasing min versions in patch releases, so this needs to reverted.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -22,13 +22,15 @@ jobs:
...
+          # pandas 1.4 requires py3.8+, remove this job on 1.4 release
+          NIGHTLY_BUILD: "false"
```
Comment (simonjayhawkins) [inline azure-pipelines.yml:27]: removing from nightly build though is fine.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @lithomas1

File: azure-pipelines.yml (hunk 1 above)
Comment (lithomas1) [reply azure-pipelines.yml:25]: 1.17.3 is the minimum version for 1.3.x though, but it wasn't bumped when the PR that released 1.3 was merged. This PR is just syncing the min versions with what is present on the main repo. See https://github.com/pandas-dev/pandas/blob/1.3.x/pyproject.toml for example which is supported by the docs link in the OP.

Comment (simonjayhawkins) [reply azure-pipelines.yml:25]: > 1.17.3 is the minimum version for 1.3.x though

thanks. is this PR ready to merge. The failures are unrelated?

File: .travis.yml [hunk 3]
```diff
@@ -25,8 +25,8 @@ jobs:
     - os: linux
       env:
         - MB_PYTHON_VERSION=3.7
-        - NP_BUILD_DEP="numpy==1.16.5"
-        - NP_TEST_DEP="numpy==1.16.5"
+        - NP_BUILD_DEP="numpy==1.17.3"
+        - NP_TEST_DEP="numpy==1.17.3"
```
Comment (simonjayhawkins) [inline .travis.yml:29]: a couple of lines below, we have `NP_BUILD_DEP="numpy==1.19.1"` for 3.8. Our min on master in pandas (i.e. 1,4) is 1.18.5. Any idea why there is a discrepancy there?

Comment (simonjayhawkins) [reply .travis.yml:29]: nvm. it's probably for the aarch support.

File: azure-pipelines.yml (hunk 1 above)
Comment (lithomas1) [reply azure-pipelines.yml:25]: Yep. Thanks for the quick reviews :).

**Comment by simonjayhawkins** [comment]
Thanks @lithomas1

**Pull request merged**